pkg/kubelet/dockershim/docker_container.go

```diff
@@ -415,7 +415,7 @@ func (ds *dockerService) ContainerStatus(_ context.Context, req *runtimeapi.Cont
 
 	labels, annotations := extractLabels(r.Config.Labels)
 	imageName := r.Config.Image
-	if len(ir.RepoTags) > 0 {
+	if ir != nil && len(ir.RepoTags) > 0 {
```

Comment:
Just to confirm... we need this check because `ir` can legitimately be `nil` when the image isn't found (as a result of the behavior on line 349)?

Reply:
Yes. When image is absent, ir would be nil.